tests/e2e/tests/content-manager/history.spec.ts

```diff
@@ -218,14 +218,25 @@ describeOnCondition(edition === 'EE')('History', () => {
       await page.getByRole('combobox', { name: 'Authors' }).click();
       await page.getByText('Will Kitman').click();
       await page.getByRole('combobox', { name: 'Authors' }).click();
+
+      // Wait for the relation response before saving, otherwise the document might save without the added relation
+      const relationResponsePromise = page.waitForResponse(async (response) => {
+        return (
+          response.status() === 200 &&
+          response.url().includes('content-manager/relations/api::article.article/authors')
+        );
+      });
       await page.getByText('Coach Beard').click();
+      await relationResponsePromise;
```

Comment:
I don't love this because it's testing an implementation detail instead of something the user will actually see. Of course I don't mind if it's needed to stop the flakiness, but I want to make sure I understand better.

This request is to populate the options of the relations picker? If so how does the test ever move past `await page.getByText('Coach Beard').click()` if the request isn't ready yet?

Reply:
You know you're right this just doesn't make sense. 

I think I had understood that when you add a relation, the draft was updated immediately with that relation. I thought we just weren't triggering a notification for that and there wasn't a way to test it using something the user sees.....but that't not the case. This was just buying us some time and probably not much different then a setTimeout sorta deal. 

I believe the problem is a relation is added but sometimes it looks like the save button is clicked so quickly that the document is saved before the relation is included in the update.

I've updated now to test the relation's link is visible before proceeding to save. Still not the most intuitive thing when reading the test but maybe it's acceptable for now 🤷‍♂️

Reply:
This makes perfect sense now 👍